cmd/query/app/grpcserver/server.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcserver
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/soheilhy/cmux"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/cmd/flags"
+	"github.com/jaegertracing/jaeger/cmd/query/app"
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+)
+
+const (
+	grpcListener = "grpc"
+	httpListener = "http"
+)
+
+// Builder provides the configuration used to build a GrpcServer
+type Builder struct {
+	Svc             *flags.Service
+	RecoveryHandler http.Handler
+	QuerySvc        querysvc.QueryService
+	Logger          *zap.Logger
+	Tracer          opentracing.Tracer
+	QueryOptions    *app.QueryOptions
```

Comment:
looks like only Port is used, I would pass the Port explicitly and narrow the API surface